# HG changeset patch
# User Rik <rik@octave.org>
# Date 1546489933 28800
#      Wed Jan 02 20:32:13 2019 -0800
# Branch stable
# Node ID aabee709c13aec7d43910a2f7b79b05c087b8849
# Parent  d0a061e6d70fba5ba652ff3dfb212e3c08d58b2c
annotation-dialog.cc: Fix static analyzer detected issues (bug #55347).

* annotation-dialog.cc (set_gui_props): Eliminate else branch on 4-elements in
"textbox" which isn't used in normal code, and could reference unitialized
memory if it did.

diff --git a/libgui/graphics/annotation-dialog.cc b/libgui/graphics/annotation-dialog.cc
--- a/libgui/graphics/annotation-dialog.cc
+++ b/libgui/graphics/annotation-dialog.cc
@@ -187,21 +187,18 @@ annotation_dialog::set_gui_props ()
               ui->sb_x->setValue (0);
               ui->sb_y->setValue (0);
             }
           if (nels >= 4)
             {
               ui->sb_width->setValue (position(2));
               ui->sb_height->setValue (position(3));
             }
-          else
-            {
-              ui->sb_width->setValue (position(2));
-              ui->sb_height->setValue (position(3));
-            }
+          // FIXME: Should there be an else branch here?
+          // In annotation.m "textbox" is forced to have a 4-elem vector.
         }
       else if (name == "string")
         {
           // FIXME: handle if is array of strings ?
           ui->edit_string->setText (props(2*i +1).string_value ().c_str ());
         }
       else if (name == "fitboxtotext")
         {
